src/plugins/media/media.js

```diff
@@ -0,0 +1,172 @@
+ *           data-media-source="…" 
+ *           data-media-type="video/…" 
+ *           data-media-controls="false" 
+ *           data-media-pauseonleave="false">
+ *      </div>
+ *
+ * Copyright 2018 Holger Teichert (@complanar)
+ * Released under the MIT license.
+ */
+/* global window, document, impress, console */
+
+(function (document, window) {
+  "use strict";
+  
+  // function names
+  var parseMediaShortcut, 
+      enhanceMediaNode, 
+      enhanceMedia, 
+      playOnStepenter,
+      pauseOnStepleave, 
+      onPlay, 
+      onPause, 
+      onEnded;
+
+  // Parse nodes with media class <div class="media" data-media-source="PATH" data-media-type="MIMETYPE"></div>
+  parseMediaShortcut = function (target) {
+    var data, mimeParts, controls, playonenter, pauseonleave;
+    data = target.dataset;
+    if (data.mediaType && data.mediaSource) {
+      mimeParts = data.mediaType.split('/');
+      target.classList.add('media-' + mimeParts[0]);
+      controls = data.mediaControls === "false" ? '' : ' controls';
+      playonenter = data.mediaPlayonenter === "true" || data.mediaPlayonenter === "" ? ' data-media-playonenter="true"' : '';
+      pauseonleave = data.mediaPauseonleave === "false" ? ' data-media-pauseonleave="false"' : '';
+      target.innerHTML = '<' + mimeParts[0] + controls + playonenter + pauseonleave + '><source src="' + data.mediaSource + '" type="' + data.mediaType + '"> Your Browser does not support HTML5 ' + mimeParts[0] + '.</' + mimeParts[0] + '>';
+    }
+  };
+  
+  onPlay = function (event) {
+    var type = event.target.nodeName.toLowerCase();
+    document.body.classList.add(type + "playing");
+    document.body.classList.remove(type + "paused");
+  };
+  
+  onPause = function (event) {
+    var type = event.target.nodeName.toLowerCase();
+    document.body.classList.add(type + "paused");
+    document.body.classList.remove(type + "playing");
+  };
+  
+  onEnded = function (event) {
+    var type = event.target.nodeName.toLowerCase();
+    document.body.classList.remove(type + "playing");
+    document.body.classList.remove(type + "paused");
+  };
+  
+  enhanceMediaNode = function (type) {
+    var i, id, media = document.getElementsByTagName(type);
+    for (i = 0; i < media.length; i++) {
+      // Set an id to identify each media node - used e.g. by the consoleMedia plugin
+      id = media[i].getAttribute('id');
+      if (id === undefined || id === null) {
+        media[i].setAttribute('id', 'media-' + type + '-' + i);
+      }
+      media[i].addEventListener("play", onPlay, false);
+      media[i].addEventListener("playing", onPlay, false);
+      media[i].addEventListener("pause", onPause, false);
+      media[i].addEventListener("ended", onEnded, false);
+    }
+  };
+  
+  enhanceMedia = function () {
+    var i, steps, media = document.getElementsByClassName('media');
+    for (i = 0; i < media.length; i++) {
+      parseMediaShortcut(media[i]);
+    }
+    enhanceMediaNode('audio');
```

Comment:
You don't need to handle `audio` and `video` separately. You can get a list of both of them with

    medias = document.querySelectorAll('audio, video');